modules/lnd.nix

```diff
@@ -213,6 +213,21 @@ in {
             while ! { exec 3>/dev/tcp/127.0.0.1/${toString cfg.rpcPort}; } &>/dev/null; do
               sleep 0.1
             done
+
+            invoicereadonlymacaroon=${cfg.dataDir}/invoice+readonly.macaroon
+            if [[ ! -f $invoicereadonlymacaroon ]]; then
+              echo Create invoice+readonly macaroon
+
+              ${pkgs.curl}/bin/curl -s \
+                -H "Grpc-Metadata-macaroon: $(${pkgs.xxd}/bin/xxd -ps -u -c 99999 '${mainnetDir}/admin.macaroon')" \
+                --cacert ${secretsDir}/lnd-cert \
+                -X POST \
+                -d '{"permissions":[{"entity":"info","action":"read"},{"entity":"onchain","action":"read"},{"entity":"offchain","action":"read"},{"entity":"address","action":"read"},{"entity":"message","action":"read"},{"entity":"peers","action":"read"},{"entity":"signer","action":"read"},{"entity":"invoices","action":"read"},{"entity":"invoices","action":"write"},{"entity":"address","action":"write"}]}' \
+                https://127.0.0.1:${restPort}/v1/macaroon |\
+                ${pkgs.jq}/bin/jq -c '.macaroon' | ${pkgs.xxd}/bin/xxd -p -r > "$invoicereadonlymacaroon"
+            fi
+            chown lnd: "$invoicereadonlymacaroon"
+            chmod 440 "$invoicereadonlymacaroon"
```

Comment:
There's a race condition: a different process might read the file before the permissions are restricted. TO fix it, you need to create an empty file with those permissions before you call curl.

This is exactly what [I do](https://github.com/Kixunil/cryptoanarchy-deb-repo-builder/blob/216a581f3296bfe37438f13def88a160a2f83509/lnd/gen_macaroons.sh#L30) + I also write to a temporary file to ensure no corrupted file will be present if the process crashes/power goes down midway.

Reply:
Implemented in 94fcf27